.github/workflows/tools.yaml

```diff
@@ -18,3 +18,16 @@ jobs:
         run: |
           npm install
           npm run ci
+      - working-directory: ./tools
+        run: |
+          make clean_and_generate
+      - name: Fail when generate introducing diff
+        run: |
+          git diff --exit-code
+      - name: Make sure that no breaking changes were introduced
+        run: | 
+          make tools v2-lint v2-test
```

Comment:
[q] any reason to use `v2-lint` instead of the GH action for golangci-lint?

Reply:
Good point. Swapping

Reply:
We do have global lint so this one sems redundant